components/server/src/theia-plugin/theia-plugin-service.ts

```diff
@@ -4,28 +4,18 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-import * as url from 'url';
 import { injectable, inject } from 'inversify';
-import { ResolvePluginsParams, ResolvedPlugins, TheiaPlugin, PreparePluginUploadParams, InstallPluginsParams, UninstallPluginParams, ResolvedPluginKind } from '@gitpod/gitpod-protocol';
+import { TheiaPlugin } from '@gitpod/gitpod-protocol';
 import { TheiaPluginDB, UserStorageResourcesDB } from "@gitpod/gitpod-db/lib";
 import { Config } from '../config';
 import { log } from '@gitpod/gitpod-protocol/lib/util/logging';
 import { ResponseError } from 'vscode-jsonrpc';
 import { ErrorCodes } from '@gitpod/gitpod-protocol/lib/messaging/error';
-import { PluginIndexEntry } from '@gitpod/gitpod-protocol/lib/theia-plugins';
 import { StorageClient } from '../storage/storage-client';
 import {  } from '@gitpod/gitpod-db/lib';
-import fetch from 'node-fetch';
-
-const builtinExtensions: PluginIndexEntry[] = require('@gitpod/gitpod-protocol/data/builtin-theia-plugins.json');
 
 const userPluginsUri = 'user-plugins://';
 
-export interface ResolvedPluginsResult {
-    resolved: ResolvedPlugins
-    external: string[]
-}
-
 @injectable()
 export class TheiaPluginService {
```

Comment:
No idea other functions should be deleted or not. Should we change the whole class name and file name/path to something not containing `theia`?

Reply:
Could you list places where it is still used?

We still have data for user settings/extensions which should be migrated to VS Code, so if they are accessed from user data sync then we should keep it. But there should not be way to generate new records, so most likely most can be removed.

Reply:
Only here use `theia-plugin-service` now

https://github.com/gitpod-io/gitpod/blob/0d05dcf813d70d5ae4678e69f61b7bd12ad1ffe8/components/server/src/code-sync/code-sync-service.ts#L176

Reply:
maybe can be inlined into code-sync endpoint to get rid of this file completely?